src/Service/VendorService/TheMovieDatabase/TheMovieDatabaseApiService.php

```diff
@@ -0,0 +1,212 @@
+{
+    private const SEARCH_URL = 'https://api.themoviedb.org/3/search/movie';
+    private const BASE_IMAGE_PATH = 'https://image.tmdb.org/t/p/original';
+
+    private $apiKey;
+    private $client;
+    private $logger;
+
+    /**
+     * TheMovieDatabaseApiService constructor.
+     *
+     * @param string                      $apiKey
+     * @param \GuzzleHttp\ClientInterface $httpClient
+     * @param \Psr\Log\LoggerInterface    $logger
+     */
+    public function __construct(string $apiKey, ClientInterface $httpClient, LoggerInterface $logger)
+    {
+        $this->apiKey = $apiKey;
+        $this->client = $httpClient;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Search in the movie database for a poster url by title, year and director.
+     *
+     * @param string $title
+     *   The title of the item
+     * @param string $originalYear
+     *   The release year of the item
+     * @param string|null $director
+     *   The director of the movie
+     *
+     * @return string
+     *   The poster url or null
+     *
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function searchPosterUrl(string $title = null, string $originalYear = null, string $director = null): ?string
+    {
+        $posterUrl = null;
+
+        // Bail out if the required information is not supplied.
+        if (null === $title || null === $originalYear || null === $director) {
+            return $posterUrl;
+        }
+
+        try {
+            $query = [
+                'query' => [
+                    'query' => $title,
+                    'year' => $originalYear,
+                    'api_key' => $this->apiKey,
+                    'page' => '1',
+                    'include_adult' => 'false',
+                    'language' => 'da_DK',
+                ],
+            ];
+
+            $responseData = $this->sendRequest(self::SEARCH_URL, $query);
+
+            $result = $this->getResultFromSet($responseData->results, $title, $director);
+
+            if ($result) {
+                $posterUrl = $this->getPosterUrl($result);
+            }
+        } catch (\Exception $e) {
+            $posterUrl = null;
+        }
+
+        return $posterUrl;
+    }
+
+    /**
+     * Get the first match in the result set.
+     *
+     * @param array $results
+     *   Array of search results
+     * @param string $title
+     *   The title of the item
+     * @param string $director
+     *   The director of the item
+     *
+     * @return \stdClass|null
+     *   The matching result or null
+     */
+    private function getResultFromSet(array $results, string $title, string $director): ?\stdClass
+    {
+        $chosenResult = null;
+
+        foreach ($results as $result) {
+            // Validate title againt result->title or result->original_title.
+            if (strtolower($result->title) === strtolower($title) || strtolower($result->original_title) === strtolower($title)) {
```

Comment:
Was it here an php normalizer would make sens to use to get around different char sets?

Or https://www.php.net/manual/en/function.mb-strtolower.php

Reply:
mb_strtolower